src/aggregate/aggregate_request.c

```diff
@@ -813,6 +818,12 @@ static ResultProcessor *buildGroupRP(PLN_GroupStep *gstp, RLookup *srclookup, Qu
 static ResultProcessor *pushRP(AREQ *req, ResultProcessor *rp, ResultProcessor *rpUpstream) {
   rp->upstream = rpUpstream;
   rp->parent = &req->qiter;
+  if (IsProfile(req)) {
+    ResultProcessor *rpProfile = RPProfile_New(NULL, NULL, 0);
```

Comment:
`RPProfile_New` is creating something out of thin air. Why not write a meaningful constructor?

Reply:
Added comment